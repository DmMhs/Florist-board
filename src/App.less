.App {
  width: 100%;
  text-align: center;
}

#toggle {
  position: absolute;
  top: 0;
  right: 0;
  margin: 10px 25px;
  font-size: 2.5rem;
  z-index: 10;
  color: #333333;
  &:hover,
  &.active {
    color: tomato;
    cursor: pointer;
  }
}
@media screen and (max-width: 576px) {
  #toggle {
    margin: 10px 20px;
  }
}
@media screen and (min-width: 576.1px) {
  #toggle {
    display: none;
  }
}