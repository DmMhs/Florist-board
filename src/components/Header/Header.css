.Header .info span {
    margin: 7px;
}