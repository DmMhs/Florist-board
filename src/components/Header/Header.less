.Header {
  .info {
    font-style: italic;
    span {
      margin: 7px;
    }
  }
}
