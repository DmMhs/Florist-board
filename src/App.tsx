import React, { Component, RefObject } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import './App.less';
import Header from './components/Header/Header';
import Shop from './components/Shop/Shop';
import Gallery from './components/Gallery/Gallery';
import Contacts from './components/Contacts/Contacts';
import Home from './components/Home/Home';
import Toggle from './components/Toggle/Toggle';
import ProductDetails from './components/ProductList/ProductDetails/ProductDetails';
import Auth from './components/Auth/Auth';
import AuthContextProvider from './components/Auth/AuthContext';

interface AppProps {}
interface AppState {
  showNavigation: boolean;
  togglePosition: string;
  products: any;
}

let resizeListener: EventListener;

class App extends Component<AppProps, AppState> {
  private toggleRef: RefObject<HTMLDivElement>;
  constructor(props: AppProps) {
    super(props);
    this.state = {
      showNavigation: window.innerWidth > 576,
      togglePosition: 'absolute',
      products: []
    };
    this.toggleRef = React.createRef();
  }

  componentDidMount() {
    if (localStorage.floristAuthLogin === undefined) {
      localStorage.setItem('floristAuthLogin', '');
    }
    if (localStorage.floristAuthToken === undefined) {
      localStorage.setItem('floristAuthToken', '');
    }
    if (localStorage.floristAuthUserId === undefined) {
      localStorage.setItem('floristAuthUserId', '');
    }
    if (localStorage.floristAuthMethod === undefined) {
      localStorage.setItem('floristAuthUserId', '');
    }
    resizeListener = () => {
      if (window.innerWidth < 768) {
        if (this.toggleRef.current!.classList.contains('active')) {
          this.toggleRef.current!.classList.remove('active');
        }
        this.setState({
          showNavigation: false,
          togglePosition: 'absolute'
        });
      } else {
        this.setState({
          showNavigation: true,
          togglePosition: 'absolute'
        });
      }
    };
    window.addEventListener('resize', resizeListener);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', resizeListener);
  }

  toggleClickedHandler = (event: React.MouseEvent<HTMLDivElement>) => {
    if (!(event.target as HTMLDivElement).classList.contains('active')) {
      this.setState({
        showNavigation: !this.state.showNavigation,
        togglePosition: 'fixed'
      });
    } else {
      this.setState({
        showNavigation: !this.state.showNavigation,
        togglePosition: 'absolute'
      });
    }
    (event.target as HTMLDivElement).classList.toggle('active');
  };

  render() {
    return (
      <BrowserRouter>
        <AuthContextProvider>
          <div className="App">
            <Toggle
              click={this.toggleClickedHandler}
              style={{
                position:
                  this.state.togglePosition === 'absolute'
                    ? 'absolute'
                    : 'fixed'
              }}
              ref={this.toggleRef}
            />
            {this.state.showNavigation ? <Header /> : null}
            {localStorage.floristAuth !== undefined
              ? JSON.parse(localStorage.floristAuth).email
              : null}
            <Switch>
              <Route path="/" exact component={Home} />
              <Route path="/shop" component={Shop} />
              <Route path="/gallery" component={Gallery} />
              <Route path="/contacts" component={Contacts} />
              <Route path="/product-details/:id" component={ProductDetails} />
              <Route path="/auth/:mode" component={Auth} />
            </Switch>
          </div>
        </AuthContextProvider>
      </BrowserRouter>
    );
  }
}

export default App;
