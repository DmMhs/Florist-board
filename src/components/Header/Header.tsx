import React from 'react';

import Navigation from './Navigation/Navigation';
import './Header.less';

const header = () => {
  return (
    <div className="Header">
      <p className="info">
        <span>FLORIST.UA</span>
        <span>MARKET SQUARE, LVIV, UA</span>
        <span>23523</span>
        <span>+123-45-678</span>
      </p>
      <Navigation />
    </div>
  );
};

export default header;
