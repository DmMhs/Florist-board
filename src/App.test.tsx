import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Header from './components/Header/Header';
import App from './App';
import { BrowserRouter } from 'react-router-dom';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('showNavigation is defined', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  expect(instance.state.showNavigation).toBeDefined();
});

it('change of showNavigation changes Header displaying', () => {
  const wrapper = shallow(<App />);
  const instance = wrapper.instance();
  instance.setState({ showNavigation: false });
  if (instance.state.showNavigation === false) {
    expect(wrapper.contains(<Header />)).toBeFalsy();
  }
  instance.setState({ showNavigation: true });
  if (instance.state.showNavigation === true) {
    expect(wrapper.contains(<Header />)).toBeTruthy();
  }
});

it('reacts on toggle click', () => {
  const wrapper = mount(<App />);
  const instance = wrapper.instance();
  instance.setState({
    showNavigation: false
  });
  wrapper.find('.Toggle').simulate('click');
  expect(instance.state.showNavigation).toBeTruthy();
  wrapper.find('.Toggle').simulate('click');
  expect(instance.state.showNavigation).toBeFalsy();
});

it('resize event has impact on state', () => {
  const wrapper = mount(<App />);
  const instance = wrapper.instance();

  instance.setState({
    showNavigation: undefined
  });

  console.log(wrapper.ref('toggleRef'));

  // global.innerWidth = 600;
  // window.dispatchEvent(new Event('resize'));
  // expect(instance.state.showNavigation).toBeFalsy();

  // global.innerWidth = 800;
  // window.dispatchEvent(new Event('resize'));
  // expect(instance.state.showNavigation).toBeTruthy();
});
