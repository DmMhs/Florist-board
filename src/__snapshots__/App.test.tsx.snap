// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App works as expected matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <App>
            <div
              className="App"
            >
              <ForwardRef
                click={[Function]}
                style={
                  Object {
                    "position": "absolute",
                  }
                }
              >
                <i
                  className="fas fa-ellipsis-h Toggle"
                  onClick={[Function]}
                  style={
                    Object {
                      "position": "absolute",
                    }
                  }
                />
              </ForwardRef>
              <header>
                <div
                  className="Header"
                >
                  <withRouter(Navigation)>
                    <Route>
                      <Navigation
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                      >
                        <div
                          className="Navigation"
                        >
                          <ul>
                            <li
                              onClick={[Function]}
                            >
                              <NavLink
                                exact={true}
                                to="/"
                              >
                                <Route
                                  exact={true}
                                  path="\\\\/"
                                >
                                  <Link
                                    aria-current="page"
                                    className="active"
                                    style={Object {}}
                                    to="/"
                                  >
                                    <a
                                      aria-current="page"
                                      className="active"
                                      href="/"
                                      onClick={[Function]}
                                      style={Object {}}
                                    >
                                      HOME
                                       
                                      <i
                                        className="fas fa-home"
                                      />
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                            <li
                              onClick={[Function]}
                            >
                              <NavLink
                                to="/shop"
                              >
                                <Route
                                  path="\\\\/shop"
                                >
                                  <Link
                                    aria-current={null}
                                    to="/shop"
                                  >
                                    <a
                                      aria-current={null}
                                      href="/shop"
                                      onClick={[Function]}
                                    >
                                      SHOP
                                       
                                      <i
                                        className="fas fa-leaf"
                                      />
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                            <li
                              onClick={[Function]}
                            >
                              <NavLink
                                to="/gallery"
                              >
                                <Route
                                  path="\\\\/gallery"
                                >
                                  <Link
                                    aria-current={null}
                                    to="/gallery"
                                  >
                                    <a
                                      aria-current={null}
                                      href="/gallery"
                                      onClick={[Function]}
                                    >
                                      GALLERY
                                       
                                      <i
                                        className="far fa-image"
                                      />
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                            <li
                              onClick={[Function]}
                            >
                              <NavLink
                                to="/contacts"
                              >
                                <Route
                                  path="\\\\/contacts"
                                >
                                  <Link
                                    aria-current={null}
                                    to="/contacts"
                                  >
                                    <a
                                      aria-current={null}
                                      href="/contacts"
                                      onClick={[Function]}
                                    >
                                      CONTACTS
                                       
                                      <i
                                        className="fas fa-map-marker-alt"
                                      />
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </li>
                            <li
                              className="accountLink"
                            >
                              <a
                                className="dropbtn"
                                onClick={[Function]}
                              >
                                MY ACCOUNT
                                 
                                <i
                                  className="fas fa-user-circle"
                                />
                                 
                                <i
                                  className="fas fa-caret-down"
                                />
                              </a>
                              <div
                                className="dropdown-content"
                              >
                                <div>
                                  <NavLink
                                    className="signup-link"
                                    onClick={[Function]}
                                    to="/auth/signup"
                                  >
                                    <Route
                                      path="\\\\/auth\\\\/signup"
                                    >
                                      <Link
                                        aria-current={null}
                                        className="signup-link"
                                        onClick={[Function]}
                                        to="/auth/signup"
                                      >
                                        <a
                                          aria-current={null}
                                          className="signup-link"
                                          href="/auth/signup"
                                          onClick={[Function]}
                                        >
                                          SIGN UP
                                           
                                          <i
                                            className="fas fa-user-plus"
                                          />
                                        </a>
                                      </Link>
                                    </Route>
                                  </NavLink>
                                  <NavLink
                                    className="signin-link"
                                    onClick={[Function]}
                                    to="/auth/signin"
                                  >
                                    <Route
                                      path="\\\\/auth\\\\/signin"
                                    >
                                      <Link
                                        aria-current={null}
                                        className="signin-link"
                                        onClick={[Function]}
                                        to="/auth/signin"
                                      >
                                        <a
                                          aria-current={null}
                                          className="signin-link"
                                          href="/auth/signin"
                                          onClick={[Function]}
                                        >
                                          SIGN IN
                                           
                                          <i
                                            className="fas fa-sign-in-alt"
                                          />
                                        </a>
                                      </Link>
                                    </Route>
                                  </NavLink>
                                </div>
                              </div>
                            </li>
                            <li
                              className="langLink"
                            >
                              <a
                                className="lang-btn"
                                onClick={[Function]}
                              >
                                <span>
                                  LANG
                                </span>
                                <img
                                  src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Flang%2Funited-kingdom.png?alt=media&token=791aa1e1-7dc6-467f-a716-3eb8dce0b313"
                                />
                                 
                                <i
                                  className="fas fa-caret-down"
                                />
                              </a>
                              <div
                                className="lang-options"
                              >
                                <a
                                  onClick={[Function]}
                                >
                                  <span>
                                    МОВА
                                  </span>
                                  <img
                                    src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Flang%2Fukraine.png?alt=media&token=9305aa3b-0e62-411e-b6ae-7dbbcaa72e74"
                                  />
                                </a>
                              </div>
                            </li>
                          </ul>
                        </div>
                      </Navigation>
                    </Route>
                  </withRouter(Navigation)>
                </div>
              </header>
              <Switch>
                <Route
                  component={[Function]}
                  computedMatch={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  exact={true}
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  path="/"
                >
                  <Home
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div
                      className="Home"
                    >
                      <WithStyles(CircularIndeterminate)>
                        <CircularIndeterminate
                          classes={
                            Object {
                              "progress": "CircularIndeterminate-progress-1",
                            }
                          }
                        >
                          <div
                            className="Spinner"
                          >
                            <WithStyles(CircularProgress)
                              className="CircularIndeterminate-progress-1"
                            >
                              <CircularProgress
                                className="CircularIndeterminate-progress-1"
                                classes={
                                  Object {
                                    "circle": "MuiCircularProgress-circle-8",
                                    "circleDisableShrink": "MuiCircularProgress-circleDisableShrink-11",
                                    "circleIndeterminate": "MuiCircularProgress-circleIndeterminate-10",
                                    "circleStatic": "MuiCircularProgress-circleStatic-9",
                                    "colorPrimary": "MuiCircularProgress-colorPrimary-5",
                                    "colorSecondary": "MuiCircularProgress-colorSecondary-6",
                                    "indeterminate": "MuiCircularProgress-indeterminate-4",
                                    "root": "MuiCircularProgress-root-2",
                                    "static": "MuiCircularProgress-static-3",
                                    "svg": "MuiCircularProgress-svg-7",
                                  }
                                }
                                color="primary"
                                disableShrink={false}
                                size={40}
                                thickness={3.6}
                                value={0}
                                variant="indeterminate"
                              >
                                <div
                                  className="MuiCircularProgress-root-2 MuiCircularProgress-colorPrimary-5 MuiCircularProgress-indeterminate-4 CircularIndeterminate-progress-1"
                                  role="progressbar"
                                  style={
                                    Object {
                                      "height": 40,
                                      "width": 40,
                                    }
                                  }
                                >
                                  <svg
                                    className="MuiCircularProgress-svg-7"
                                    viewBox="22 22 44 44"
                                  >
                                    <circle
                                      className="MuiCircularProgress-circle-8 MuiCircularProgress-circleIndeterminate-10"
                                      cx={44}
                                      cy={44}
                                      fill="none"
                                      r={20.2}
                                      strokeWidth={3.6}
                                      style={Object {}}
                                    />
                                  </svg>
                                </div>
                              </CircularProgress>
                            </WithStyles(CircularProgress)>
                          </div>
                        </CircularIndeterminate>
                      </WithStyles(CircularIndeterminate)>
                      <div
                        className="slider-wrapper"
                      >
                        <Slider
                          auto={true}
                          bannerModeEnabled={true}
                          images={Array []}
                          showControls={true}
                        >
                          <div
                            className="Slider"
                          >
                            <div
                              className="slides-wrapper"
                              style={
                                Object {
                                  "transform": "translateX(0px)",
                                  "transition": "transform ease-out 1s",
                                }
                              }
                            />
                            <LeftArrow
                              goToPrevSlide={[Function]}
                              show={true}
                            >
                              <div
                                className="left-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "display": "flex",
                                  }
                                }
                              >
                                <i
                                  aria-hidden="true"
                                  className="fas fa-chevron-left"
                                />
                              </div>
                            </LeftArrow>
                            <RightArrow
                              goToNextSlide={[Function]}
                              show={true}
                            >
                              <div
                                className="right-arrow"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "display": "flex",
                                  }
                                }
                              >
                                <i
                                  aria-hidden="true"
                                  className="fas fa-chevron-right"
                                />
                              </div>
                            </RightArrow>
                          </div>
                        </Slider>
                      </div>
                    </div>
                  </Home>
                </Route>
              </Switch>
              <footer>
                <div
                  className="Footer"
                >
                  <span>
                    FLORIST.UA © All Rights Reserved
                  </span>
                </div>
              </footer>
            </div>
          </App>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
